### ISSUE TITLE
Spell out your plans (to make money)

### ISSUE THREAD

**Issue by finn-matti** [pieces-app/documentation#114]
The only thing I've found so far is this closed issue #11 

>In regards to pricing, Pieces is currently free and does not cost a thing - there may be some opportunity to pay for faster features in the future as well. :)

Please be more up front with how you intend to make money. How you're currently financing this pursuit and what we can expect your plans to look like in, say, five years. What is the pricing going to look like: How much per month/year? Because you want us to pour in a lot of love by curating, tagging, etc. our snippets, I'd love to know more than what I can expect now.

**Comment by jordan-pieces** [comment]
Hey @finn-matti thanks for reaching out and to answer your questions:

**All of our current features will be available to existing users at no cost, forever.**

Our main focus right now is to continue building a product that makes the day-to-day development experience simpler, and we have not yet monetized in any way. That being said, we are starting to reach out to enterprise organizations to initiate paid pilots that will extend our runway as a startup. 

As for individuals, beyond the free forever plan, we will likely have options to upgrade your cloud storage, access new features that haven't yet been released, and increase the number of messages you can send to the Pieces Copilot with pricing ideally lower than a monthly Netflix subscription. 

For now, [we always appreciate users buying us a coffee](https://www.buymeacoffee.com/pieces)!

We haven't exactly defined this model yet but **can promise that your personal Pieces repo will always remain accessible and free** - and we’re always open to feedback on the pricing we are thinking about down below.

Additionally, we really appreciate you joining Pieces at this early stage - and we want to make sure we reward early adopters for helping us to work through the kinks in our own development journey. To give thanks, you and other early adopters will receive a discount on our upgraded plans once they launch, including the ability to be grandfathered into new upcoming product updates for a set period.

<img width="1920" alt="9" src="https://github.com/pieces-app/documentation/assets/55861512/a37a6686-8649-4262-9e42-f7e0c1bffb43">

**Comment by tsavo-at-pieces** [comment]
Hey @finn-matti - Tsavo here 👋 I'm the CEO & Technical Co-Founder at Pieces 🙃

First and foremost, I want to say thank you for giving our product a go. We seriously appreciate every one of our early adopters, and the feedback they give us is critical in refining our Product Market Fit.

Regarding the discussion above:

Candidly, we've been so busy focusing on engineering, product, and enterprise that we haven't prioritized payments infrastructure to date. It's a bit intimidating, especially since we are an offline, on-device native app, and also written entirely in Dart – two areas where Stripe and others haven't prioritized drop-in solutions 😅. 

Because of that, we've been committing as many engineering resources as possible to build a product that is absolutely indispensable. In investor speak, that roughly means a complete focus on key PLC metrics, i.e., DAU/MAU/WAU + L28, 3mo & 6mo retentions.

I will say this: one of the reasons we've been able to keep Pieces for Developers free, without compromising on a loaded feature set, is thanks to the fact that 90% of the system in single-player mode is entirely on-device.

... aaannnddd the last 10% that is currently cloud-powered (the Copilot Chat) has a MAJOR update with several on-device and hardware-accelerated options coming to the LLM runtime: 7Bn, 13Bn, and 70Bn-Params. This is scheduled to release next Friday, August 18th 🎉.

TL;DR: On-Device means we aren't paying massive cloud costs or having to host user data.

Worst comes to worst, say we don't exist in 5 years (highly unlikely, but we'll roll with it for this scenario 😅), our current and upcoming features will all still be able to run on your device if the year is 2028 🙌. Ironically, the feature set will likely be faster than ever considering the upcoming chip advancements from Apple, Intel, Qualcomm, etc. 🤝.

Further, we have a data export functionality coming up, which will enable you to export/import/transfer your saved materials fluidly between formats like JetBrains Snippets, VS Code Snippets, Markdown, JSON, CSV, etc. Believe it or not, some of this functionality is already available today in our VS Code Plugin!

As for the cloud features, our main goal is to simply usher/sync data between devices/teams rather than store everything in a centralized, vendor-locked server farm.

This is actually the reason why we require users to provision your own cloud, _optionally with a custom subdomain_, to leverage our current link sharing functionality. We intentionally took a decentralized approach to our product's cloud capabilities to enable self-hosting early in 2024 (very important for our enterprise customers).

Long story short - we're a bunch of privacy-aware developers (myself included) who hate vendor lock-in, and we continuously push to ensure our product provides users with authentic flexibility.

Let me know your take on all this! At the end of the day, we're moving fast and figuring it out with people such as yourself and our community at large 😃.

I'd love to connect, and you can reach me personally any time at [tsavo@pieces.app](mailto:tsavo@pieces.app).

Cheers & Have a great weekend!
-Tsavo
